src/share/util/shr_mem_mod.F90

```diff
@@ -48,25 +48,25 @@ subroutine shr_mem_init(prt, strbuf)
     endif
 
     ierr = GPTLget_memusage (msize, mrss0, mshare, mtext, mdatastack)
-    allocate(mem_tmp(1024*1024))    ! 1 MWord, 8 MB
+    allocate(mem_tmp(1024*1024*128))    ! 1 MWord, 1024 MB (128*8)
```

Comment:
I would like to at least know the reasoning behind this change.   A namelist option might be a good idea if there is in fact value in making this larger on some systems.

Reply:
This was done in https://github.com/E3SM-Project/E3SM/pull/3361

@amametjanov , can you comment here?

Reply:
This is only done on one master process to estimate memory page size, which can vary from one machine to another. 8MB allocate was too small on `Anvil` (Linux CentOS7 cluster).

Reply:
@amametjanov And if a machine does not have 1GB of memory?  Maybe that is quite rare but it may be good to at least explicitly capture an allocate error here and provide an appropriate message.

Reply:
Also, are there performance concerns with allocating and then deallocating this much memory, in terms of the time it takes to do the allocation, memory fragmentation, etc.? (I really don't know much about this, so just asking naively.)

I'm also wondering what the implications are for very small machines. We make the point that CESM can be run on a laptop, at least for some configurations (single point, or coarse-resolution global grids for testing, etc.). These days most laptops probably have more than 1 GB of memory, but if a lot of that is already used for the OS, etc., what will happen here?

@amametjanov In addition to @jedwards4b 's recent comment about catching the allocate error, can you also comment on the idea of leaving the default at 8 MB, but allowing that to be overridden on a machine-by-machine basis? I'm just concerned about doing something that might be convenient for developers on a few systems but at the same time may cause problems for users on other systems.

Reply:
Thanks for backing this out, @amametjanov